### DISCUSSION TITLE
RenderSection not working

### DISCUSSION THREAD

**Discussion by JoshClose** [statiqdev/Discussions#52] [Q&A]
In my layout page I have
```cshtml
@RenderSection("Scripts", required: false)
```
and in my header partial I have this at the bottom
```cshtml
@section Scripts
{
	<script>
		document.getElementById("navbar-burger").addEventListener("click", (ev) => {
			ev.currentTarget.classList.toggle("is-active");
		});
	</script>
}
```
The script isn't being rendered.

I'm including the partial in the layout like this
```cshtml
<partial name="_Header" />
```

Am I doing something wrong here?

**Comment by daveaglick** [comment]
Two things come to mind:
- Does the layout bring the default tag helpers in scope (including `partial`) with `@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers`? This can also be in a `_ViewImports.cshtml`. Since tag helpers look like HTML when not in scope, you don't get an error if they're not available and it just fails silently. Another way to test is to put some other non-section content in the `_Header.cshtml` file and see if it renders or check the generated HTML via devtools to see if the `<partial...>` tag is still in the HTML or if it was replaced by the tag helper.
- I honestly didn't know you could replace a `@section` from inside a partial. I thought sections could only be defined in the page and it's layouts. Does this work in "normal" Razor?

**Comment by JoshClose** [comment]
Everything looks as it should.

> I honestly didn't know you could replace a @section from inside a partial. I thought sections could only be defined in the page and it's layouts. Does this work in "normal" Razor?

lol, I'm not sure about this one. Just looked it up and it's not possible. That's too bad. I can just throw it in the layout instead.

**Reply by daveaglick** [reply]
Bummer! That would have been pretty cool if you could :).
